lighthouse-core/driver/index.js

```diff
@@ -100,15 +100,19 @@ class Driver {
     const loadData = {};
     let pass = Promise.resolve();
 
+    if (config.traceName) {
+      loadData[config.traceName] = {};
```

Comment:
Re top-level: I had that originally, but we are accumulating the `loadData` later using `Object.assing()`, meaning the individual traces would overwrite each other.

Your other suggestions sounds like a much better solution, will give that a spin.